addons/csw/functions/fnc_reload_loadMagazine.sqf

```diff
@@ -19,46 +20,32 @@
  * Public: No
  */
 
-params ["_vehicle", "_turret", "_carryMag", "_magSource", "_unit"];
-TRACE_5("loadMagazine",_vehicle,_turret,_carryMag,_magSource,_unit);
+params ["_vehicle", "_turret", "_carryMag", "_magSource", "_unit", "_ammo"];
+TRACE_6("loadMagazine",_vehicle,_turret,_carryMag,_magSource,_unit,_ammo);
 
-private _timeToLoad = 1;
-if (!isNull(configOf _vehicle >> QUOTE(ADDON) >> "ammoLoadTime")) then {
-    _timeToLoad = getNumber(configOf _vehicle >> QUOTE(ADDON) >> "ammoLoadTime");
-};
+private _timeToLoad = GET_NUMBER(configOf _vehicle >> QUOTE(ADDON) >> "ammoLoadTime", 1);
 
-private _displayName = format [localize LSTRING(loadX), getText (configFile >> "CfgMagazines" >> _carryMag >> "displayName")];
+private _displayName = format [LLSTRING(loadX), getText (configFile >> "CfgMagazines" >> _carryMag >> "displayName")];
 
 private _onFinish = {
-    (_this select 0) params ["_vehicle", "_turret", "_carryMag", "_magSource", "_unit"];
-    TRACE_5("load progressBar finish",_vehicle,_turret,_carryMag,_magSource,_unit);
+    (_this select 0) params ["_vehicle", "_turret", "_carryMag", "_magSource", "_unit", "_ammo"];
+    TRACE_6("load progressBar finish",_vehicle,_turret,_carryMag,_magSource,_unit,_ammo);
 
-    ([_vehicle, _turret, _carryMag, _magSource] call FUNC(reload_canLoadMagazine)) params ["", "", "_neededAmmo", ""];
+    ([_vehicle, _turret, _carryMag] call FUNC(reload_canLoadMagazine)) params ["", "", "_neededAmmo", ""];
     if (_neededAmmo <= 0) exitWith { ERROR_1("Can't load ammo - %1",_this); };
 
-    // Figure out what we can add from the magazines we have
-    private _bestAmmoToSend = -1;
-    {
-        _x params ["_xMag", "_xAmmo"];
-        if (_xMag == _carryMag) then {
-            if ((_bestAmmoToSend == -1) || {(_xAmmo > _bestAmmoToSend) && {_xAmmo <= _neededAmmo}}) then {
-                _bestAmmoToSend = _xAmmo;
-            };
-        };
-    } forEach (if (_magSource isKindOf "CAManBase") then {magazinesAmmo _magSource} else {magazinesAmmoCargo _magSource});
+    [_magSource, _carryMag, _ammo] call EFUNC(common,removeSpecificMagazine);
```

Comment:
`EFUNC(common,removeSpecificMagazine)` returns if the removal was successful or not. I'm tempted to say that we should utilise that to ensure that the magazine was correctly removed - or do you think it's unnecessary, because of `FUNC(reload_canLoadMagazine)` a lot of checks?

Reply:
Unnecessary.